src/HomeWorkLvl3/Lesson5/gasStation/FuelStation.java

```diff
@@ -0,0 +1,49 @@
+package HomeWorkLvl3.Lesson5.gasStation;
+
+import HomeWorkLvl3.Lesson5.cars.Cars;
+
+import java.util.concurrent.Semaphore;
+
+
+public class FuelStation {
+    private GasPool gasPool;
+    public static final int CAR_RESTRICTION = 3;
+    private Semaphore semaphore;
+
+    public FuelStation() {
+        this.gasPool = new GasPool();
+        this.semaphore = new Semaphore(CAR_RESTRICTION);
+    }
+
+    public boolean isEnoughFuel(float amount) {
+        return gasPool.getFuelReserveGP() - amount >= 0;
+    }
+
+    public float refuelTheCar(float amount, Cars car) {
+        try {
+            semaphore.acquire(3);
+            System.out.printf("[%s] прибыл на заправку%n", car.getStringID());
+            System.out.printf("[%s] Запрос топлива в количестве %.1f, на складе %.1f \n",
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [FS: Format string should use %n rather than n (VA_FORMAT_STRING_USES_NEWLINE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#fs-format-string-should-use-n-rather-than-n-va-format-string-uses-newline).
This format string includes a newline character (\n). In format strings, it is generally  preferable to use %n, which will produce the platform-specific line separator.